src/declarations/stencil-public-compiler.ts

```diff
@@ -2192,30 +2192,268 @@ export type OutputTarget =
+   * URL, this value will be appended to the URL and that will be checked for
+   * a precache match. This should be set to what your web server is using for
+   * its directory index.
+   *
+   * @default 'index.html'
+   */
+  directoryIndex?: string | undefined;
+
+  /**
+   * Assets that match this will be assumed to be uniquely versioned via their
+   * URL, and exempted from the normal HTTP cache-busting that's done when
+   * populating the precache. While not required, it's recommended that if your
+   * existing build process already inserts a `[hash]` value into each filename,
+   * you provide a RegExp that will detect that, as it will reduce the bandwidth
+   * consumed when precaching.
+   */
+  dontCacheBustURLsMatching?: RegExp | undefined;
+
+  /**
+   * Determines whether or not symlinks are followed when generating the precache
+   * manifest. For more information, see the definition of `follow` in the `glob`
+   * [documentation](https://github.com/isaacs/node-glob#options).
+   *
+   * @default true
+   */
+  globFollow?: boolean | undefined;
+
+  /**
+   * A set of patterns matching files to always exclude when generating the
+   * precache manifest. For more information, see the definition of `ignore` in the `glob`
+   * [documentation](https://github.com/isaacs/node-glob#options).
+   *
+   * @default ['node_modules/**']
+   */
   globIgnores?: string | string[];
-  templatedUrls?: any;
-  maximumFileSizeToCacheInBytes?: number;
-  manifestTransforms?: any;
-  modifyUrlPrefix?: any;
-  dontCacheBustURLsMatching?: RegExp;
-  navigateFallback?: string;
-  navigateFallbackWhitelist?: RegExp[];
-  navigateFallbackBlacklist?: RegExp[];
-  cacheId?: string;
-  skipWaiting?: boolean;
-  clientsClaim?: boolean;
-  directoryIndex?: string;
-  runtimeCaching?: any[];
-  ignoreUrlParametersMatching?: any[];
-  handleFetch?: boolean;
+
+  /**
+   * Files matching any of these patterns will be included in the precache
+   * manifest. For more information, see the
+   * [`glob` primer](https://github.com/isaacs/node-glob#glob-primer).
+   *
+   * @default ['**.{js,css,html}']
+   */
+  globPatterns?: string[] | undefined;
+
+  /**
+   * If true, an error reading a directory when generating a precache manifest
+   * will cause the build to fail. If false, the problematic directory will be
+   * skipped. For more information, see the definition of `strict` in the `glob`
+   * [documentation](https://github.com/isaacs/node-glob#options).
+   *
+   * @default true
+   */
+  globStrict?: boolean | undefined;
+
+  /**
+   * Any search parameter names that match against one of the RegExp in this array
+   * will be removed before looking for a precache match. This is useful if your
+   * users might request URLs that contain, for example, URL parameters used to
+   * track the source of the traffic.
+   *
+   * @default [/^utm_/]
+   */
+  ignoreURLParametersMatching?: RegExp[] | undefined;
```

Comment:
Similar to my comment above, it looks like we narrow this type from `any[]` to `RegExp[]`, which we might want to consider a breaking change as well